### ISSUE TITLE
Create function code action generates functions with compilation error

### ISSUE THREAD

**Issue by gayalkuruppu** [ballerina-platform/ballerina-lang#34656]
**Description:**
1.

https://user-images.githubusercontent.com/46120162/149284576-35ee2000-14d2-4cde-9870-58fbc5f8eb3e.mov

2.

https://user-images.githubusercontent.com/46120162/149284589-ca272937-b8a7-468f-a346-6bab3d8f71bb.mov

3.

https://user-images.githubusercontent.com/46120162/149284610-55c7fc7e-a553-41c2-a428-8ef4cf5d9734.mov

**Steps to reproduce:**
1.
```
type ER1 error<record {| ER1 e?; |}>;

public function main() {
    ER1 er1 = getER1();
}
```
2.
```
type T [T, T]|int|string;

public function main() {
    T x = getT();
}
```
3.
```
type Never never;
type Function function(int);
type FN Function|never;

public function main() {
    FN x = getFN();
}
```
**Affected Versions:**
SL 2201.0.0

**OS, DB, other environment details and versions:**
MacOS Big Sur

**Comment by nadeeshaan** [comment]
For case 1, I tried the following fix

```
case ERROR:
                if (depth > MAX_DEPTH) {
                    return Optional.of("");
                }
                TypeSymbol errorType = CommonUtil.getRawType(((ErrorTypeSymbol) rawType).detailTypeDescriptor());
                StringBuilder errorString = new StringBuilder("error (\"\"");
                if (errorType.typeKind() == TypeDescKind.RECORD) {
                    List<RecordFieldSymbol> mandatoryFields = getMandatoryRecordFields((RecordTypeSymbol) errorType);
                    if (!mandatoryFields.isEmpty()) {
                        errorString.append(", ");
                        errorString.append(mandatoryFields.stream()
                                .map(recordFieldSymbol -> recordFieldSymbol.getName().get()
                                        + " = " + getDefaultValueForType(recordFieldSymbol.typeDescriptor(), depth + 1)
                                        .orElse(""))
                                .collect(Collectors.joining(", ")));
                    }
                }
                errorString.append(")");
                typeString = errorString.toString();
                break;
```
This might be one option

**Comment by IMS94** [comment]
* Case 1 can be fixed easily as @nadeeshaan suggested. 
* Case 2 is a bit complex. due to recursive nature of the type.
* For case 3, we had a similar requirement and decided to not address now due to the complexity of generating a default value to a function. (see https://github.com/ballerina-platform/ballerina-lang/pull/34447#issuecomment-1004514830 comment)

**Comment by malinthar** [comment]
Discussion tracked on https://docs.google.com/document/d/1X7j-Z6bFH80_CdOoxtIOKc8Rwl4WL6rXsd3l4_HLPRU/edit?usp=sharing

**Comment by nadeeshaan** [comment]
> * Fix create function code action issues #34447 (comment)

@IMS94 let's create a separate issue to address use-case 3. It is better to keep it as a separate known issue.
